website/src/blogs/how-to-deploy-a-cassandra-cluster-ring-on-kubernetes-openebs.md

```diff
@@ -0,0 +1,453 @@
+
+    apiVersion: apps/v1beta1
+     kind: StatefulSet
+     metadata:
+     name: cassandra
+     labels:
+     app: cassandra
+     spec:
+     serviceName: cassandra
+     replicas: 3
+     selector:
+     matchLabels:
+     app: cassandra
+     template:
+     metadata:
+     labels:
+     app: cassandra
+     spec:
+     containers:
+     — name: cassandra
+     image: gcr.io/google-samples/cassandra:v11
+     imagePullPolicy: Always
+     ports:
+     — containerPort: 7000
+     name: intra-node
+     — containerPort: 7001
+     name: tls-intra-node
+     — containerPort: 7199
+     name: jmx
+     — containerPort: 9042
+     name: cql
+     resources:
+     limits:
+     cpu: “500m”
+     memory: 1Gi
+     requests:
+     cpu: “500m”
+     memory: 1Gi
+     securityContext:
+     capabilities:
+     add:
+     — IPC_LOCK
+     lifecycle:
+     preStop:
+     exec:
+     command: [“/bin/sh”, “-c”, “PID=$(pidof java) && kill $PID && while ps -p $PID > /dev/null; do sleep 1; done”]
+     env:
+     — name: MAX_HEAP_SIZE
+     value: 512M
+     — name: HEAP_NEWSIZE
+     value: 100M
+     — name: CASSANDRA_SEEDS
+     value: “cassandra-0.cassandra.default.svc.cluster.local”
+     — name: CASSANDRA_CLUSTER_NAME
+     value: “K8Demo”
+     — name: CASSANDRA_DC
+     value: “DC1-K8Demo”
+     — name: CASSANDRA_RACK
+     value: “Rack1-K8Demo”
+     — name: CASSANDRA_AUTO_BOOTSTRAP
+     value: “false”
+     — name: POD_IP
+     valueFrom:
+     fieldRef:
+     fieldPath: status.podIP
+     readinessProbe:
+     exec:
+     command:
+     — /bin/bash
+     — -c
+     — /ready-probe.sh
+     initialDelaySeconds: 15
+     timeoutSeconds: 5
+     # These volume mounts are persistent. They are like inline claims,
+     # but not exactly because the names need to match exactly one of
+     # the stateful pod volumes.
+     volumeMounts:
+     — name: cassandra-data
+     mountPath: /cassandra_data
+     volumeClaimTemplates:
+     — metadata:
+     name: cassandra-data
+     annotations:
+     volume.beta.kubernetes.io/storage-class: openebs-cassandra
+     spec:
+     accessModes: [ “ReadWriteOnce” ]
+     resources:
+     requests:
+     storage: 5G
+
+Note: There are few parameters you may want to modify.
+
+`apiVersion: apps/v1beta2` API group and version is introduced 1.8 release.
+
+`replicas: 3`, i’m starting with 3 replicas and will increase later.
+
+`image: gcr.io/google-samples/cassandra:v12` is the latest image available at the time I&#8217;ve tested.
+
+`volume.beta.kubernetes.io/storage-class: openebs-cassandra` i’m using a predefined OpenEBS storage class. You can modify it separately.
+
+#### Create a Cassandra Headless Service
+
+To be able to have a simple discovery of the Cassandra seed node we need to create a “headless” service. If you view the `cassandra-service.yaml`file, you will notice that `clusterIP` is set to None. This will allows us to use KubeDNS for the Pods to discover the IP address of the Cassandra seed.
```

Comment:
```suggestion
To be able to have a simple discovery of the Cassandra seed node we need to create a “headless” service. If you view the `cassandra-service.yaml`file, you will notice that `clusterIP` is set to None. This will allow us to use KubeDNS for the Pods to discover the IP address of the Cassandra seed.
```